src/hybrid-appliance/azext_hybrid_appliance/custom.py

```diff
@@ -0,0 +1,172 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import os
+import psutil
+import requests
+import subprocess
+import yaml
+from knack.log import get_logger
+from azure.cli.core import get_default_cli
+
+from subprocess import Popen, PIPE, run, STDOUT, call, DEVNULL
+from azext_hybrid_appliance import _constants as consts
+from azure.cli.core.azclierror import ValidationError, CLIInternalError
+logger = get_logger(__name__)
+
+
+def validate_hybrid_appliance(resource_group_name, name):
+
+    all_validations_passed = True
+
+    # Check if the operating system is Linux
+    if not psutil.LINUX:
+        all_validations_passed = False
+        logger.warning("This program can only run on a Linux machine")
+
+    # Check if the disk space is at least 20GB
+    disk_usage = psutil.disk_usage('/')
+    if disk_usage.total < consts.Disk_Threshold * 1024 * 1024 * 1024:
+        all_validations_passed = False
+        logger.warning("This program requires at least {} of disk space".format(consts.Disk_Threshold))
+
+    # Check if the memory is at least 4GB
+    memory = psutil.virtual_memory()
+    if memory.total < consts.Memory_Threshold * 1024 * 1024 * 1024:
+        all_validations_passed = False
+        logger.warning("This program requires at least {} of memory".format(consts.Memory_Threshold))
+    
+    # Check if pre-req endpoints are reachable
+    endpoints = [consts.Snap_Config_Storage_Endpoint, consts.Apt_Pull_Public_Endpoint, consts.Snap_Pull_Public_Endpoint, consts.App_Insights_Endpoint]
+
+    for endpoint in endpoints:
+        try:
+            response = requests.head(endpoint, timeout=5)
+            response.raise_for_status()
+        except (requests.exceptions.RequestException):
+            all_validations_passed = False
+            logger.warning("The endpoint {} is not reachable from your machine".format(endpoint))
+    
+    try:
+        cmd_show_arc= ['az', 'connectedk8s', 'show', '-n', name, '-g', resource_group_name, '-o', 'none']
+        process = subprocess.Popen(cmd_show_arc)
+        if process.wait() == 0:
+            print("The appliance name and resource group name passed already correspond to an existing connected cluster. Please try again with a different appliance name.")
```

Comment:
Nit: Should we use the word "connected cluster". CX wouldn't know what is means right?